ir/base.def

```diff
@@ -32,18 +32,25 @@ abstract Type {
     virtual const Type* getP4Type() const = 0;
 }
 
-/// Implemented by all types that may be generic
+/// Implemented by all types that may be generic:
+/// Control, Extern, Method, Package, and Parser
 interface IMayBeGenericType {
+    /// Allows the retrieval of type parameters
     virtual const TypeParameters* getTypeParameters() const = 0;
-    const Type* toType() const { return to<Type>(); }
+    // cast to a Type object.
+    // There is a single invocation of this method (typeChecker.cpp:92) so removed
+    // const Type* toType() const { return to<Type>(); }
 }
 
-/// Implemented by objects with an 'apply' method
+/// Implemented by objects with an 'apply' method: Parser, Control, Table
+/// \TODO: Unclear why the corresponding Type objects of these objects also implement IApply
```

Comment:
We don't really use inheritance for IR classes; only abstract classes and interfaces are subclassed.
In other words, today all IR concrete classes are final.
Maybe we could use inheritance, I don't know if there are technical reasons.